src/pages/workspaces/workspace/Data.js

```diff
@@ -556,6 +558,10 @@ const ToolDrawer = _.flow(
   ])
 })
 
+const SnapshotContent = ({ snapshotDetails, snapshotKey }) => {
+  return snapshotDetails === undefined ? spinnerOverlay : h(SnapshotInfo, { snapshotId: snapshotDetails[snapshotKey[0]].id, snapshotName: snapshotKey[0] })
```

Comment:
Why is this a separate component, rather than just having this ternary in the cond below? And why are we passing in the whole snapshotKey array if we're only ever using the first element, rather than just passing `selectedDataType[0]` below?

Reply:
This is a separate component because we will also need to add the snapshot table here, along with the snapshot info. I can change it to just pass in the first thing in the array (the snapshot name), but the next ticket (adding tables) will require using the second item in the array (the table name). So you'll just have to change it back anyway

Reply:
Ah, cool. In that case, try this:
```suggestion
const SnapshotContent = ({ snapshotDetails, snapshotKey: [snapshotName] }) => {
  return snapshotDetails === undefined ? spinnerOverlay : h(SnapshotInfo, { snapshotId: snapshotDetails[snapshotName].id, snapshotName })
```

I would still recommend moving the spinner down to the cond, since that's in the component in charge of loading, and this shouldn't be rendered until there's something for it to show.

Reply:
looks like it renders the spinner even when I take the spinnerOverlay stuff out from here!